src/query/procedure/mg_procedure_impl.cpp

```diff
@@ -1695,7 +1701,14 @@ mgp_error mgp_vertex_copy(mgp_vertex *v, mgp_memory *memory, mgp_vertex **result
   return WrapExceptions([v, memory] { return NewRawMgpObject<mgp_vertex>(memory, *v); }, result);
 }
 
-void mgp_vertex_destroy(mgp_vertex *v) { DeleteRawMgpObject(v); }
+void mgp_vertex_destroy(mgp_vertex *v) {
+  if (v->graph->ctx && v->graph->ctx->auth_checker &&
+      !v->graph->ctx->auth_checker->Accept(*v->graph->ctx->db_accessor, v->impl, v->graph->view /*, CREATE_DELETE*/)) {
+    return;
```

Comment:
Do we throw an exception?

Reply:
No, we will expand return mgp_error enum and return that value